src/testrender/shading.cpp

```diff
@@ -1446,6 +1452,114 @@ struct MxSheen final : public BSDF, MxSheenParams {
     }
 };
 
+// Implement the sheen model proposed in:
+//  "Practical Multiple-Scattering Sheen Using Linearly Transformed Cosines"
+// Tizian Zeltner, Brent Burley, Matt Jen-Yuan Chiang - Siggraph 2022
+// https://tizianzeltner.com/projects/Zeltner2022Practical/
+struct ZeltnerBurleySheen final : public BSDF, MxSheenParams {
+    ZeltnerBurleySheen(const MxSheenParams& params)
+        : BSDF(), MxSheenParams(params)
+    {
+    }
+
+#define USE_LTC_SAMPLING 1
+
+    Color3 get_albedo(const Vec3& wo) const override
+    {
+        const float NdotV = clamp(N.dot(wo), 0.0f, 1.0f);
+        return Color3(fetch_ltc(NdotV).z);
+    }
+
+    Sample eval(const Vec3& wo, const Vec3& wi) const override
+    {
+        const Vec3 L = wi, V = wo;
+        const float NdotV = clamp(N.dot(V), 0.0f, 1.0f);
+        const Vec3 ltc    = fetch_ltc(NdotV);
+
+        const Vec3 T1     = (V - N * NdotV).normalize();
+        const Vec3 T2     = N.cross(T1);
+        const Vec3 localL = Vec3(T1.dot(L), T2.dot(L), N.dot(L));
+
+        const float aInv = ltc.x, bInv = ltc.y, R = ltc.z;
+        Vec3 wiOriginal(aInv * localL.x + bInv * localL.z, aInv * localL.y,
+                        localL.z);
+        const float len2 = dot(wiOriginal, wiOriginal);
+
+        float det      = aInv * aInv;
+        float jacobian = det / (len2 * len2);
+
+#if USE_LTC_SAMPLING == 1
+        float pdf = jacobian * std::max(wiOriginal.z, 0.0f) * float(M_1_PI);
+        return { wi, Color3(R), pdf, 1.0f };
+#else
+        float pdf = float(0.5 * M_1_PI);
+        // NOTE: sheen closure has no fresnel/masking
+        return { wi, Color3(2 * R * jacobian * std::max(wiOriginal.z, 0.0f)),
+                 pdf, 1.0f };
+#endif
+    }
+
+    Sample sample(const Vec3& wo, float rx, float ry, float rz) const override
+    {
+#if USE_LTC_SAMPLING == 1
+        const Vec3 V      = wo;
+        const float NdotV = clamp(N.dot(V), 0.0f, 1.0f);
+        const Vec3 ltc    = fetch_ltc(NdotV);
+        const float aInv = ltc.x, bInv = ltc.y, R = ltc.z;
+        Vec3 wi;
+        float pdf;
+        Sampling::sample_cosine_hemisphere(Vec3(0, 0, 1), rx, ry, wi, pdf);
+
+        wi = Vec3(wi.x - wi.z * bInv, wi.y, wi.z * aInv).normalize();
+
+        const Vec3 T1 = (V - N * NdotV).normalize();
+        const Vec3 T2 = N.cross(T1);
+        const Vec3 L  = Vec3(wi.x * T1 + wi.y * T2 + wi.z * N);
+
+        // TODO: is there a cheaper way to get the jacobian here?
+        const Vec3 wiOriginal(aInv * wi.x + bInv * wi.z, aInv * wi.y, wi.z);
+        const float len2     = dot(wiOriginal, wiOriginal);
+        const float det      = aInv * aInv;
```

Comment:
Here's what I'm doing, using the length of L / wi instead (before normalisation): https://github.com/AcademySoftwareFoundation/MaterialX/blob/main/libraries/pbrlib/genglsl/lib/mx_microfacet_sheen.glsl#L169-L183

Reply:
Nice - I'll go through this and try to consolidate the logic.  Didn't realize yours was added to the MaterialX side already.

Reply:
I think if you retain the `aInv` multiplication of `wi` but otherwise follow what I'm doing then you'll end up with `square(len2 / aInv)` for the jacobian, which should still be a little bit cheaper than what you have here.

NB: I may end up changing to fitting to work with the non-inverse matrix coefficients (TBD, but it's possible that they could be more linear), but that shouldn't change the implementation all that much.